File: src/SimpleTodo/Services/TodoDto.cs
namespace SimpleTodo.Services;

public sealed class CreateTodoDto
{
    public string Title { get; set; }
    public string Description { get; set; }
}

public sealed class UpdateTodoDto
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
}

public sealed class TodoDto
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime? DoneAt { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: src/simpletodoui/src/apiClients/models/todoDto.ts
export class TodoDto {
    id: string;
    title: string;
    description: string;
    doneAt?: Date;
    createdAt: Date;

    constructor(
        id: string,
        title: string,
        description: string,
        createdAt: Date,
        doneAt?: Date
    ) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.doneAt = doneAt;
        this.createdAt = createdAt;
    }
}
